### PR TITLE
Add scala rules support to buck project

### PR DESCRIPTION
**Pull request by kageiit**

### CODE REVIEW

**Comment by kageiit** [comment]
cc @styurin

**Comment by kageiit** [comment]
Ping on this :)
@styurin

**Comment by kageiit** [comment]
@styurin Pushed after rebase and running google java format. Please take a look

File: src/com/facebook/buck/jvm/scala/ScalaLibraryDescription.java [hunk 1]
```diff
@@ -97,23 +88,7 @@ public void findDepsForTargetFromConstructorArgs(
     Optionals.addIfPresent(scalaBuckConfig.getScalacTarget(), extraDepsBuilder);
   }
 
-  // Note: scala does not have a exported_deps because scala needs the transitive closure of
-  // dependencies to compile. deps is effectively exported_deps.
-  interface CoreArg extends CommonDescriptionArg, HasDeclaredDeps, HasSrcs, HasTests {
-    @Value.NaturalOrder
-    ImmutableSortedSet<SourcePath> getResources();
-
-    ImmutableList<String> getExtraArguments();
```
Comment (styurin) [inline src/com/facebook/buck/jvm/scala/ScalaLibraryDescription.java]: This is not present in `JavaLibraryDescription.CoreArg`

**Comment by styurin** [comment]
Do you want to an integration test?

File: src/com/facebook/buck/jvm/scala/ScalaLibraryDescription.java (hunk 1 above)
Comment (kageiit) [inline src/com/facebook/buck/jvm/scala/ScalaLibraryDescription.java]: Good catch. Let me add it

**Comment by kageiit** [comment]
> Do you want to an integration test?

Sure I will add one

**Comment by kageiit** [comment]
@styurin reran everything through google java format and added an integration test

**Comment by kageiit** [comment]
Rebased to remove conflict

**Comment by styurin** [comment]
Build is failing, check the log in travis.

File: src/com/facebook/buck/ide/intellij/BUCK [hunk 2]
```diff
@@ -50,6 +50,7 @@ java_immutables_library(
         "//src/com/facebook/buck/rules:types",
         "//src/com/facebook/buck/rules/coercer:interface",
         "//src/com/facebook/buck/rules/keys:keys",
+        "//src/com/facebook/buck/jvm/scala:scala",
```
Comment (styurin) [inline src/com/facebook/buck/ide/intellij/BUCK]: looks like wrong place for this dep, can you run buildifier?

Comment (kageiit) [inline src/com/facebook/buck/ide/intellij/BUCK]: Done

File: src/com/facebook/buck/ide/intellij/BUCK [hunk 3]
```diff
@@ -37,6 +37,7 @@ java_immutables_library(
         "//src/com/facebook/buck/jvm/java:config",
         "//src/com/facebook/buck/jvm/java:packagefinder",
         "//src/com/facebook/buck/jvm/java:rules",
+        "//src/com/facebook/buck/jvm/scala:scala",
```
Comment (styurin) [inline src/com/facebook/buck/ide/intellij/BUCK]: still the wrong place :)

Comment (kageiit) [inline src/com/facebook/buck/ide/intellij/BUCK]: oops. Updated now! :)

**Comment by styurin** [comment]
Tests are failing in travis, need to add checks for Scala presence?

**Comment by kageiit** [comment]
Finally! :)

**Pull request closed without merging**